talktorials/5_compound_clustering/T5_compound_clustering.ipynb

```diff
@@ -142,7 +142,7 @@
        "        "
```

Comment:
<p>Looking at the plot, I was first wondering about the gaps. That might be just the rendering?</p><p>Then I thought, well, now that the cluster sizes are ordered, we can get away with a simple line plot, right?</p><p> </p><p>Also when looking at the code, I am wondering:</p><ol><li>Move the actual plotting step (<code>plot()</code>) to the top, then do the plot cosmetics (setting axis, labels, ...).</li><li>Initialize <code>Figure</code> and <code>Axes</code>  object in one line (instead of two lines).</li><li>Currently, the <code>Axes</code>  is named <code>plt1</code>,  however the convention is (I think) to name it <code>ax</code>  - also I suggest to do the plot cosmetics on the <code>Axes</code>  object directly (<code>ax.set_xlabel()</code>  instead of <code>plt.xlabel()</code> ).</li><li>X-axis should start at 1 (not 0), since our cluster index starts at 1.</li><li>I do not see why cut off one cluster at the end:<code>plt1.bar(range(1, len(clusters)), [len(c) for c in clusters[:len(clusters)-1]], lw=0)</code></li></ol><p> </p><p>So I suggest to replace the cell's code as follows:</p><p> </p><p><code># Plot the size of the clusters</code></p><p><code>fig, ax = plt.subplots(nrows=1, ncols=1, figsize=(10, 4))  </code></p><p><code>ax.plot(range(1, len(clusters)+1), [len(c) for c in clusters])  </code></p><p><code>ax.set_xlim([1, len(clusters)])  </code></p><p><code>ax.set_ylim([0, len(clusters[0])+1])  </code></p><p><code>ax.set_xlabel('Cluster index', fontsize=20)  </code></p><p><code>ax.set_ylabel('Number of molecules', fontsize=20)  </code></p><p><code>ax.tick_params(labelsize=16)  </code></p><p><code>plt.show()  </code></p><p> </p><p>(The code formatting with ReviewNB looks weird, maybe I am doing something wrong here, sorry.)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/TeachOpenCADD/pull/38/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>